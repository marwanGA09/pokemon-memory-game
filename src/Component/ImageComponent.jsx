import { IsLoading } from './IsLoading';

export function ImageComponent({ img, isLoading, onSelectImage }) {
  return (
    <>
      {isLoading ? (
        <IsLoading />
      ) : (
        <img
          src={img.image}
          alt="pokemon card"
          onClick={() => onSelectImage(img.id)}
        />
      )}
    </>
  );
}
