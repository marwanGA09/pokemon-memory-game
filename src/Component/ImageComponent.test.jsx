import { render, screen } from '@testing-library/react';
import { describe, test, expect, vitest } from 'vitest';
import { ImageComponent } from './ImageComponent';
import userEvent from '@testing-library/user-event';

describe('Image component', () => {
  const imageObject = {
    id: 1,
    image:
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
    selected: false,
  };

  test('Presence of Pokemon image component in the document if isLoading is false', () => {
    render(<ImageComponent img={imageObject} isLoading={false} />);
    const imageElem = screen.getByAltText(/pokemon/i);
    console.log(imageElem.src);
    expect(imageElem).toMatchSnapshot();
    expect(imageElem.src).toBe(imageObject.image);
  });

  test('Presence of IsLoading component in the document if isLoading is true', () => {
    render(<ImageComponent img={imageObject} isLoading={true} />);
    const imageElem = screen.getByAltText(/spinner/i);
    expect(imageElem).toMatchSnapshot();
    expect(imageElem.src).not.toBe(imageObject.image);
  });

  test('calling OnselectImage with correct argument', async () => {
    const onSelectImage = vitest.fn();
    const userInteraction = userEvent.setup();
    render(
      <ImageComponent
        img={imageObject}
        isLoading={false}
        onSelectImage={onSelectImage}
      />
    );
    const imageElem = screen.getByAltText(/pokemon/i);
    await userInteraction.click(imageElem);
    expect(onSelectImage).toBeCalled();
    expect(onSelectImage).toBeCalledWith(imageObject.id);
    expect(onSelectImage).toBeCalledTimes(1);
  });
});
