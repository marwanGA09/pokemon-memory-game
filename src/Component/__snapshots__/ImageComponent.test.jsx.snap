// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image component > Presence of IsLoading component in the document if isLoading is true 1`] = `
<img
  alt="spinner gif image"
  src="Spinner.gif"
/>
`;

exports[`Image component > Presence of Pokemon image component in the document if isLoading is false 1`] = `
<img
  alt="pokemon card"
  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
/>
`;

exports[`Image component > Presence of Pokemon image component in the document if isLoading is false 2`] = `
<img
  alt="pokemon card"
  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
/>
`;
